darts/utils/historical_forecasts/optimized_historical_forecasts_regression.py

```diff
@@ -33,20 +37,34 @@ def _optimized_historical_forecasts_last_points_only(
     show_warnings: bool = True,
     verbose: bool = False,
     predict_likelihood_parameters: bool = False,
+    data_transformers: Optional[Dict[str, Pipeline]] = None,
     **kwargs,
 ) -> Union[TimeSeries, Sequence[TimeSeries], Sequence[Sequence[TimeSeries]]]:
     """
     Optimized historical forecasts for RegressionModel with last_points_only = True
 
     Rely on _check_optimizable_historical_forecasts() to check that the assumptions are verified.
     """
+    if data_transformers is None:
+        data_transformers = dict()
     forecasts_list = []
     iterator = _build_tqdm_iterator(series, verbose)
     for idx, series_ in enumerate(iterator):
         past_covariates_ = past_covariates[idx] if past_covariates is not None else None
         future_covariates_ = (
             future_covariates[idx] if future_covariates is not None else None
         )
+        # Pipeline/DataTransformer must already be fitted, transform everything in one go
+        if len(data_transformers) > 0:
```

Comment:
I think the transformation should be applied on all series together, and not individually.

Also, in my opinion, we can move the transform / inverse transform out of this function and into `ForecastingModel.historical_forecasts()`. Apply transform before calling `model_optimized_historical_forecasts()`, and inverse transform before returning.

That would simplify things quite a bit

Reply:
I thought about it, decided to make it redundant in case some people want to call the optimized routine directly but I don't mind moving most of the logic to `historical_forecasts` and have only the inverse transform occur in the optimized methods.

Reply:
I would say we can leave the optimized routines naive to transformations. Also, applying it on all series/hist fc output is more efficient due to parallelization :)